repository-data/webfiles/src/main/resources/site/freemarker/hee/macros/tabs.ftl

```diff
@@ -0,0 +1,25 @@
+<#assign hst=JspTaglibs["http://www.hippoecm.org/jsp/hst/core"]>
+
+<#macro tabs tabs>
+    <#if tabs??>
+        <#if tabs.tabsContentBlock.tabPanel?size gt 1>
+            <#assign randPref = tabs.tabsContentBlock.randomPrefix>
+            <div class="nhsuk-tabs" data-module="nhsuk-tabs">
+                <h2 class="nhsuk-tabs__title">${tabs.tabsContentBlock.title}</h2>
+                <div class="nhsuk-tabs__list" role="tablist" aria-label="${tabs.tabsContentBlock.title}" data-module="nhsuk-tabs">
+                    <#list tabs.tabsContentBlock.tabPanel as tabPanel>
+                        <button class="nhsuk-tabs__list-item${(tabPanel?is_first?then(' nhsuk-tabs__list-item--selected',''))}" role="tab" aria-selected="true"

```

Comment:
I think the tab index would make the tab elements `id, aria-controls, etc` unique. In that case, could we get rid of `randPref ` from the template as well as from `site/components/src/main/java/uk/nhs/hee/web/beans/Tabs.java` unless it will have an impact on the component functionality ?

Reply:
Hi @smanivasag - I may have misunderstood what you mean by tab index, but rand prefix has been added to handle scenario where more than one tab components have been added to the page. Otherwise we could potentially end up with two elements having the same id.

Reply:
Ah Ok, that makes perfect sense - Is it OK if we leverage tabs block identifier for unique/random prefix ?
From:

```
<#assign randPref = tabs.tabsContentBlock.randomPrefix>
```
To:

```
<#assign randPref = tabs.identifier?keep_before('-')>
```

Reply:
Thanks for suggestion. Just had to prefix it with a letter.

Reply:
Great, thanks 👍